TrendMicroEmailSec/trendmicro_modules/trigger_email_sec.py

```diff
@@ -0,0 +1,270 @@
+import time
+import traceback
+import urllib.parse
+from functools import cached_property
+from threading import Event, Lock, Thread
+from typing import Generator
+
+import orjson
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+from sekoia_automation.storage import PersistentJSON
+
+from . import TrendMicroModule
+from .client import ApiClient
+from .helpers import iso8601_to_timestamp, unixtime_to_iso8601
+from .metrics import FORWARD_EVENTS_DURATION, INCOMING_MESSAGES, OUTCOMING_EVENTS
+
+
+class TrendMicroConnectorConfiguration(DefaultConnectorConfiguration):
+    frequency: int = 60
+    batch_size: int = 500
+
+
+class TrendMicroWorker(Thread):
+    API_VERSION = "v1"
+
+    def __init__(self, connector: "TrendMicroEmailSecurityConnector", log_type: str):
+        super().__init__()
+
+        self.log_type = log_type
+        self.connector = connector
+
+        self.service_url = connector.module.configuration.service_url
+        if self.service_url.startswith("http://"):
+            self.service_url = "https://%s" % self.service_url[7:]
+
+        elif not self.service_url.startswith("https://"):
+            self.service_url = "https://%s" % self.service_url
+
+        self._stop_event = Event()
+
+    def stop(self):
+        self._stop_event.set()
+
+    @property
+    def running(self):
+        return not self._stop_event.is_set()
+
+    @property
+    def frequency(self):
+        return self.connector.configuration.frequency
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(
+            username=self.connector.module.configuration.username, api_key=self.connector.module.configuration.api_key
+        )
+
+    def get_last_timestamp(self) -> int:
+        now = int(time.time())  # in seconds
+
+        self.connector.context_lock.acquire()
+        with self.connector.context as cache:
+            most_recent_ts_seen = cache.get(self.log_type, {}).get("last_timestamp")
+        self.connector.context_lock.release()
+
+        # if undefined, retrieve events from the last 5 minutes
+        if most_recent_ts_seen is None:
+            return now - 5 * 60
+
+        # We can't retrieve messages older than 72 hours
+        one_week_ago = now - 72 * 60 * 60
+        if most_recent_ts_seen < one_week_ago:
+            most_recent_ts_seen = one_week_ago
+
+        return most_recent_ts_seen
```

Comment:
Why do you use timestamp? `datetime` wasn't suitable?

Reply:
We can use `datetime` and `timedelta` here and anywhere in the connector as well. But IMO it's just more concise and aligned with other connectors this way